src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs

```diff
@@ -5152,6 +5255,31 @@ private static float Reduce(Func<float, float, float> reduceOp, float[] op1)
             return acc;
         }
 
+        public static float AddAcrossRecursivePairwise(float[] op1) => ReduceRecursivePairwise(Add, op1);
+
+        private static float ReduceRecursivePairwise(Func<float, float, float> reduceOp, float[] op1)
+        {
+            if (op1.Length == 2)
+            {
+                return reduceOp(op1[0], op1[1]);
+            }
+
+            if (op1.Length % 2 != 0)
```

Comment:
do we have input parameters that exercise this condition?

Reply:
No. This would require an vector where the number of elements was not a power of 2.

I wasn't sure if there was a way of raising an exception in the testing. So instead, NaN would ensure the test failed.